### PR TITLE
Add recceiver id, which is used by cleanOnStart and cleanOnStop

### PR DESCRIPTION
**Pull request by jacomago**

### CODE REVIEW

File: server/recceiver/cfstore.py [hunk 1]
```diff
@@ -521,6 +512,15 @@ def __updateCF__(proc, pvInfoByName, delrec, hostName, iocName, iocid, owner, io
...
 
+def create_properties(owner, iocTime, channels_dict, iocs, recceiver_id, ch):
+    return [
+                        {u'name': 'hostName', u'owner': owner, u'value': iocs[channels_dict[ch[u'name']][-1]]["hostname"]},
+                        {u'name': 'iocName', u'owner': owner, u'value': iocs[channels_dict[ch[u'name']][-1]]["iocname"]},
+                        {u'name': 'iocid', u'owner': owner, u'value': channels_dict[ch[u'name']][-1]},
+                        {u'name': 'pvStatus', u'owner': owner, u'value': 'Active'},
+                        {u'name': 'time', u'owner': owner, u'value': iocTime},
+                        {u'name': RECCEIVER_ID_KEY, u'owner': owner, u'value': recceiver_id}]
+
```
Comment (simon-ess) [inline server/recceiver/cfstore.py]: While I do really like these cleanups, it might have been better to put them in a separate commit to make it clear what is cleanup, and what is the functional change.

File: server/recceiver/cfstore.py [hunk 2]
```diff
@@ -467,7 +457,8 @@ def __updateCF__(proc, pvInfoByName, delrec, hostName, iocName, iocid, owner, io
                      {u'name': 'iocName', u'owner': owner, u'value': iocName},
                      {u'name': 'iocid', u'owner': owner, u'value': iocid},
                      {u'name': 'pvStatus', u'owner': owner, u'value': "Active"},
-                     {u'name': 'time', u'owner': owner, u'value': iocTime}]
+                     {u'name': 'time', u'owner': owner, u'value': iocTime},
+                     {u'name': RECCEIVER_ID_KEY, u'owner': owner, u'value': recceiver_id}]
```
Comment (simon-ess) [inline server/recceiver/cfstore.py]: Given your new function `create_properties` has a lot of overlap with this, isn't it possible to use it here as well?

File: server/recceiver/cfstore.py [hunk 3]
```diff
@@ -224,7 +227,7 @@ def _commitWithThread(self, TR):
         pvInfoByName = {}
         for rid, (info) in pvInfo.items():
             if info["pvName"] in pvInfoByName:
-                _log.warn("Commit contains multiple records with PV name: %s (%s)", pv, iocid)
+                _log.warn("Commit contains multiple records with PV name: %s (%s)", info["pvName"], iocid)
```
Comment (simon-ess) [inline server/recceiver/cfstore.py:231]: Good catch...

**Review by simon-ess — COMMENTED** [review]
Broadly speaking this seems like a good change. My one main concern is that leaving it up to the user to configure the RECCEIVER_ID seems potentially problematic. What happens if multiple recceivers share a key? Is there no way to use something like `hostname` or something similar to specify the ID?

File: server/recceiver/cfstore.py (hunk 1 above)
Comment (jacomago) [reply server/recceiver/cfstore.py]: done

File: server/recceiver/cfstore.py (hunk 2 above)
Comment (jacomago) [reply server/recceiver/cfstore.py]: done

**Comment by tynanford** [comment]
This is a good idea and would address https://github.com/ChannelFinder/recsync/issues/74 nicely. I agree with @simon-ess that using something like the hostname would help reduce manual config and user errors. What do you think @jacomago ?

Could you update cf.conf and demo.conf with the new config option `RECCEIVER_ID` so it's more visible to new users?

File: server/recceiver/cfstore.py [hunk 4]
```diff
@@ -502,20 +507,22 @@ def __updateCF__(proc, pvInfoByName, delrec, hostName, iocName, iocid, owner, io
     if proc.cancelled:
         raise defer.CancelledError()
 
-def create_properties(owner, iocTime, hostName, iocName, iocid):
+def create_properties(owner, iocTime, recceiver_id, hostName, iocName, iocid):
```
Comment (jeonghanlee) [inline server/recceiver/cfstore.py]: I think, we should use the same convention here between `receiver_id` and `iocid`. Please select it either convention to keep the similar style, even if we have multiple convention. I don't want to see more conventions in a code.

Comment (simon-ess) [reply server/recceiver/cfstore.py]: I think that's good advice; there are already two conventions in that line (`hostName` vs. `iocid`), adding a third is probably overkill :smile:

Comment (jacomago) [reply server/recceiver/cfstore.py]: Updated

**Comment by jacomago** [comment]
> This is a good idea and would address #74 nicely. I agree with @simon-ess that using something like the hostname would help reduce manual config and user errors. What do you think @jacomago ?
> 
> Could you update cf.conf and demo.conf with the new config option `RECCEIVER_ID` so it's more visible to new users?

Set the default value to be the hostname and updated the conf files.

**Comment by simon-ess** [comment]
LGTM. It would be nice in theory to fix the failing pipelines, but those are pretty clearly a separate task that should be handled otherwise (in particular the Centos7 ones; I haven't seen the other failures before)

**Review by simon-ess — APPROVED** [review]
